src/test/java/io/neonbee/test/base/ClusteredNeonBeeTestBase.java

```diff
@@ -0,0 +1,24 @@
+package io.neonbee.test.base;
+
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.parallel.Isolated;
+
+import io.neonbee.NeonBeeOptions.Mutable;
+
+/**
+ * The {@link ClusteredNeonBeeTestBase} is a little syntax sugar for test classes, that are dealing with clustered
+ * tests. This class essentially wraps two things together and thusly decreases the risk of making a mistake:
+ * <ol>
+ * <li>Cluster tests should always run isolated. Thus the {@link ClusteredNeonBeeTestBase} is {@link Isolated}.</li>
+ * <li>This test base overrides the {@link #adaptOptions(TestInfo, Mutable)} method, in order to set all tests to run
+ * clustered by default.</li>
+ * </ol>
+ */
+@Isolated("Clustered NeonBee tests must always run isolated. The default FakeClusterManager has a static state and multiple real cluster manager should never run side-by-side for multiple tests to avoid interfering")
+public class ClusteredNeonBeeTestBase extends NeonBeeTestBase {
```

Comment:
I don'T have a good feeling for ClusteredNeonBeeTestBase, because NeonBeeTestBase can only handle one Node .. so you will NEVER create a cluster with NeonBeeTestBase and this is what "ClusteredNeonBeeTestBase" looks like.

I would suggest to have a method like "provideUserPrincipal" to "enable" the cluster mode for this single NeonBee instance.

I then would rename ClusteredNeonBeeExtensionTestBase into ClusteredNeonBeeTestBase.

Reply:
You are right. I adapted the change as you suggested but renamed the new test base to ClusterExtensionTestBase to make it more distingushed from the NeonBeeTestBase